api/src/routes/settings.test.ts

```diff
@@ -122,6 +127,220 @@ describe('settingRoutes', () => {
+          `/confirm-email?email=${encodedEmail}&token=${validButMissingToken}`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(res.status).toBe(302);
+      });
+
+      // TODO(Post-MVP): there's no need to keep the auth token around if,
+      // somehow, the user is missing
+      it.todo(
+        'should delete the auth token if there is no user associated with it'
+      );
+
+      it('should reject requests when the email param is different from user.newEmail', async () => {
+        await fastifyTestInstance.prisma.user.update({
+          where: { id: defaultUserId },
+          data: { newEmail: 'an@oth.er' }
+        });
+
+        const res = await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validToken}`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(defaultErrorMessage)
+        );
+        expect(res.status).toBe(302);
+      });
+
+      it('should reject requests if the auth token has expired', async () => {
+        const res = await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${expiredToken}`
+        );
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' +
+            formatMessage({
+              content:
+                'The link to confirm your new email address has expired. Please try again.',
+              type: 'info'
+            })
+        );
+        expect(res.status).toBe(302);
+      });
+
+      it('should update the user email', async () => {
+        const res = await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validToken}`
+        );
+        const user = await fastifyTestInstance.prisma.user.findUniqueOrThrow({
+          where: { id: defaultUserId }
+        });
+
+        expect(res.headers.location).toBe(
+          'http://localhost:8000?' + formatMessage(successMessage)
+        );
+        expect(user.email).toBe(newEmail);
+      });
+
+      it('should clean up the user record', async () => {
+        await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validToken}`
+        );
+
+        const user = await fastifyTestInstance.prisma.user.findUniqueOrThrow({
+          where: { id: defaultUserId }
+        });
+
+        expect(user.newEmail).toBeNull();
+        expect(user.emailVerified).toBe(true);
+        expect(user.emailVerifyTTL).toBeNull();
+        expect(user.emailAuthLinkTTL).toBeNull();
+      });
+
+      it('should remove the auth token on success', async () => {
+        await superGet(
+          `/confirm-email?email=${encodedEmail}&token=${validToken}`
+        );
+
+        const authToken = await fastifyTestInstance.prisma.authToken.findUnique(
+          {
+            where: { id: validToken }
+          }
+        );
+
+        expect(authToken).toBeNull();
+      });
```

Comment:
Can we not check these in the success case's test scenario?

Reply:
Sure, but I think it's helpful to have the tests to be narrow. Then they give better feedback to whoever broke them.